pkg/trace/transform/transform.go

```diff
@@ -100,6 +102,27 @@ func OtelSpanToDDSpanMinimal(
 	return ddspan
 }
 
+func isDatadogAPMConventionKey(k string) bool {
+	return k == "service.name" || k == "operation.name" || k == "resource.name" || k == "span.type" || k == semconv127.AttributeDeploymentEnvironmentName || k == semconv.AttributeDeploymentEnvironment || k == "http.method" || k == "http.status_code"
```

Comment:
```suggestion
	return k == "service.name" || k == "operation.name" || k == "resource.name" || k == "span.type" || k == "http.method" || k == "http.status_code"
```

don't exclude deployment.environment or deployment.environment.name for now for backwards compatibility